src/postgres/postgresConnectionStrings.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionOptions, parse } from "pg-connection-string";
+import { ParsedConnectionString } from "../ParsedConnectionString";
+import { nonNullProp } from "../utils/nonNull";
+
+export async function parsePostgresConnectionString(connectionString: string): Promise<ParsedPostgresConnectionString> {
+    connectionString = connectionString ? connectionString.trim() : '';
+    let accountConnection: boolean = false;
+    if (!connectionString.match(/^postgres:\/\/[^\/\s]+\/[^\/\s]+$/)) {
+        if (connectionString.charAt(connectionString.length - 1) === "/") {
+            connectionString = connectionString + "postgres";
+        } else {
+            connectionString = connectionString + "/postgres";
+        }
+        accountConnection = true;
+    }
+    const config: ConnectionOptions = parse(connectionString);
+    return new ParsedPostgresConnectionString(connectionString, config, accountConnection);
+}
+
+export class ParsedPostgresConnectionString extends ParsedConnectionString {
+    public readonly hostName: string;
+    public readonly port: string;
+    public readonly username: string;
+    public readonly password: string;
+    public readonly accountConnection: boolean;
```

Comment:
You don't need an `accountConnection` property. `ParsedConnectionString` already states that `databaseName` is what should be used to determine if it's an account connection or not. One reason that's preferred is because it helps the compiler automatically figure out when `databaseName` is defined and means you don't have to use `nonNullProp` as much. For example, this would throw a typescript error because `string | undefined` can't be assigned to `string`:
```
if (!connectionString.accountConnection) {
    const example: string = connectionString.databaseName;
}
```

However, this would have no error:
```
if (connectionString.databaseName) {
    const example: string = connectionString.databaseName;
}
```